bosh-director/lib/bosh/director/rendered_job_templates_cleaner.rb

```diff
@@ -1,20 +1,33 @@
 module Bosh::Director
   class RenderedJobTemplatesCleaner
-    def initialize(instance_model, blobstore)
+    def initialize(instance_model, blobstore, logger)
       @instance_model = instance_model
       @blobstore = blobstore
+      @logger = logger
     end
 
     def clean
       @instance_model.stale_rendered_templates_archives.each do |archive|
-        @blobstore.delete(archive.blobstore_id)
+        if @blobstore.exists? archive.blobstore_id
```

Comment:
Let's do that:
- local seems to already handle that
- s3 is almost there except it does not raise NotFound (how come delete_object does not use AWS::S3::Errors::NoSuchKey error just like get_object?)